craft_cli/messages.py

```diff
@@ -526,9 +528,11 @@ class _PipeReaderThread(threading.Thread):
     # byte used to unblock the reading (under Windows)
     UNBLOCK_BYTE = b"\x00"
 
-    def __init__(self, printer: _Printer, stream: Optional[TextIO], use_timestamp: bool):
+    def __init__(
+        self, printer: _Printer, stream: Optional[TextIO], printer_flags: Dict[str, bool]
```

Comment:
it's just a suggestion but have you considered using a dataclass instead of a `Dict[str,bool]` for the printer flags? It's more self-documenting and protects against "wrong" dict keys

Reply:
Yes, but it would mean that everytime a message is shown/produced (for every line in the subprocess), the dict will have to be built and rebuilt to call the printer .show() method... just a dict sounded more natural for this.

Reply:
A possibility would be to take in the dataclass and convert it to a dict in `__init__`, but I'm not sure it's worth the hassle

Reply:
Right.